### PR TITLE
Add documentation for pipeline templates (EDGE-963)

### PR DESCRIPTION
**Pull request by mhaylock**
Introduces documentation for the new Pipeline Templates feature.

Per guidance from @ozdenyilmaz, this is under a new "Governance" section.

This is my first crack at a documentation update, I look forward to learning how to make this better! 🙌 

Missing:
~~- [ ] Updated GraphQL docs (should have this ready tomorrow)~~
~~- [ ] An example in the REST API docs (not sure if this is needed?)~~

### CODE REVIEW

File: config/routes.rb
Comment (mhaylock) [inline config/routes.rb:88]: I seem to be breaking from a pattern of _not_ having a redirect like this, but it seems like `/docs/governance` should be more than a 404?

File: pages/governance/pipeline_templates.md
Comment (mbelton-buildkite) [inline pages/governance/pipeline_templates.md:21]: Passive -> active voice:

```suggestion
An administrator can add multiple templates to use across the organization. Making changes and saving a template will apply those changes to all pipelines using that template.
```

File: pages/governance/pipeline_templates.md
Comment (mbelton-buildkite) [inline pages/governance/pipeline_templates.md:15]: Add a lead-in sentence for these steps:

```suggestion

To create a template:

```

File: pages/governance/pipeline_templates.md
Comment (mbelton-buildkite) [inline pages/governance/pipeline_templates.md:16]: I think we should take them right there since we have a link:

```suggestion
1. Navigate to your [organization’s pipeline template settings](https://buildkite.com/organizations/-/pipeline-templates).
```

File: pages/governance/pipeline_templates.md
Comment (mbelton-buildkite) [inline pages/governance/pipeline_templates.md:17]: Click -> Select, missing comma:

```suggestion
1. If this is your first template, select _Create a Template_. Otherwise, select _New Template_. A pipeline template is created automatically for you.
```

File: pages/governance/pipeline_templates.md
Comment (mbelton-buildkite) [inline pages/governance/pipeline_templates.md:18]: Click -> Select:

```suggestion
1. Update the name and description of your new template and select _Apply_ to save your changes.
```

File: pages/governance/pipeline_templates.md
Comment (mbelton-buildkite) [inline pages/governance/pipeline_templates.md:19]: Click -> Select:

```suggestion
1. You can also select _Edit_ to change the step configuration for the template.
```

File: pages/governance/pipeline_templates.md
Comment (mbelton-buildkite) [inline pages/governance/pipeline_templates.md:25]: General tightening, avoid "via" because it can be ambiguous:

```suggestion
An administrator can test a pipeline template against a pipeline using the _New Build_ button on the pipeline page.
```

File: pages/governance/pipeline_templates.md
Comment (mbelton-buildkite) [inline pages/governance/pipeline_templates.md:27]: Missing comma:

```suggestion
If a template exists for the organization, it can be selected from the _Pipeline template_ dropdown to create a new build using the step configuration from that template.
```

**Review by mbelton-buildkite — COMMENTED** [review]
I wasn't able to finish the review, but submitting this for now so you can think about the URL comment. I'll come back to it to do the last two sections on the Pipeline templates page and the Overview tomorrow morning.

File: data/nav.yml
Comment (mhaylock) [reply data/nav.yml:310]: @mbelton-buildkite wouldn't an equivalent of the Deployments section be something like this?:
```suggestion
    - name: "Governance"
      children:
        - name: "Overview"
          path: "governance"
        - name: "Pipeline templates"
          path: "governance/pipeline-templates"
```

Is the reason for your suggestion that you are uncertain if we will keep the "Governance" section?

I'm not too worried about what the section is used, my main concern is having a known URL we can link to from the app (we link to the page from several places).

I think I'll defer to @ozdenyilmaz for helping make a decision here, as she had proposed the URLs I've used here.

File: pages/governance/pipeline_templates.md
Comment (mhaylock) [reply pages/governance/pipeline_templates.md:10]: @mbelton-buildkite accepting this suggestion thanks - is "Buildkite dashboard" a shorthand for the whole UI that we use in the docs?

File: data/nav.yml
Comment (mbelton-buildkite) [reply data/nav.yml:310]: Ah yip, what I'm suggesting is a mix between the Deployments and Security sections. 

And yes, I'd like to avoid including "governance" in the URL because it's not clear to me that it's the right section/label. I'm happy to use it in this change, but want to make it easier for us to update later if we need to.

File: pages/governance/pipeline_templates.md
Comment (mbelton-buildkite) [reply pages/governance/pipeline_templates.md:10]: Yeah, it's a decision I inherited that I'm trying to go with. I'll probably revisit it sometime, but for now I want to see how well it fits in different places.

File: pages/governance/pipeline_templates.md
Comment (mhaylock) [reply pages/governance/pipeline_templates.md:17]: @mbelton-buildkite I'm not sure "select" is the right word here, but I can see the confusion.

The challenge I had with writing this instruction is that the page you see is entirely different depending upon whether you have created a template already or not.

So when you are yet to create a template you get:
Empty state
Where as once you have a template it looks like:
List view

Perhaps it's easiest to just describe the empty state only?

File: data/nav.yml
Comment (mbelton-buildkite) [reply data/nav.yml:310]: To help the Overview page feel more meaningful, I think it would be good to include a few more pages in the section. Are there any existing pages that would fit there?

The candidates I see are:

- [Build exports](https://buildkite.com/docs/pipelines/build-exports)
- [Build retention](https://buildkite.com/docs/pipelines/build-retention)
- [Public pipelines](https://buildkite.com/docs/pipelines/public-pipelines)

What do you think?

File: pages/governance/pipeline_templates.md
Comment (mhaylock) [reply pages/governance/pipeline_templates.md:17]: @mbelton-buildkite oh I just saw your next comment - I guess "select" is just the phrase you use for tap/click?

In that case, ignore this 😅

Comment (mbelton-buildkite) [reply pages/governance/pipeline_templates.md:17]: All good. I imagine most people are using Buildkite on a computer, but I think it's better to use the general term. This comes from [Microsoft's style guide](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/c/click).

**Comment by mbelton-buildkite** [comment]
@mhaylock, I did a bigger rework for the final two sections so have created a new branch to make it easier for you to review: #2257.

File: data/nav.yml
Comment (ozdenyilmaz) [reply data/nav.yml:310]: As far as I understand this just changes the url to be governance/templates instead of governance/pipeline-templates. I am more than ok with that change, makes sense since we're in pipelines as a section.

Comment (ozdenyilmaz) [reply data/nav.yml:310]: build exports definitely belong here, I am not sure about retention because there is a bigger piece of archiving builds.

not sure about public pipelines also

Comment (mhaylock) [reply data/nav.yml:310]: > As far as I understand this just changes the url to be governance/templates instead of governance/pipeline-templates. I am more than ok with that change, makes sense since we're in pipelines as a section.

Clarified with @ozdenyilmaz that the change is to "pipelines/templates" not "governance/templates" - but she is happy with that so I'll go ahead and make that URL change @mbelton-buildkite.

(I won't make any changes around moving other things into the "Governance" section, for now at least).

Comment (mhaylock) [reply data/nav.yml]: @mbelton-buildkite updated with [1c909b0](https://github.com/buildkite/docs/pull/2255/commits/1c909b0a5d60322cca9cf0a080d0951673a2341f)

Comment (mbelton-buildkite) [reply data/nav.yml]: All good! I've fixed the conflict and will merge into this branch when the tests pass.

I'll propose moving the Build exports page into here in another PR.

File: pages/pipelines/governance.md
Comment (mbelton-buildkite) [inline pages/pipelines/governance.md:3]: This helps the page look more worthwhile, but it doesn't say all that much:

```suggestion
Governance plays a crucial role in CI/CD tools, particularly for anyone operating in regulated industries or handling sensitive data. It encompasses policies, processes, and controls to ensure practices align with the following:

- Industry-specific regulations.
- Internal policies.
- Widely recognized compliance standards.

These standards assess the security, availability, processing integrity, confidentiality, and privacy of information systems.

Buildkite understands the importance of meeting compliance and auditing requirements. [Pipeline templates](/docs/pipelines/templates) help you control the step configurations used across pipelines in your organization.

```

What do you think?

Comment (mbelton-buildkite) [reply pages/pipelines/governance.md:3]: Note, I didn't include SOC specifically because I couldn't see an obvious link to include for it. I'm happy to put it in if you think it's important, and even happier if we can put it in with a link to an explanation.

Comment (mhaylock) [reply pages/pipelines/governance.md:3]: Thanks @mbelton-buildkite, myself and @ozdenyilmaz think this is a great improvement 👍

File: pages/pipelines/templates.md
Comment (mbelton-buildkite) [inline pages/pipelines/templates.md:4]: Nitpick: This works when you think about "Pipeline templates" as a feature name, but I think it reads more smoothly with "are." Happy for you to decide which way to go.

```suggestion
> Pipeline templates are only available on an [Enterprise](https://buildkite.com/pricing) plan.
```

File: pages/pipelines/templates.md
Comment (mbelton-buildkite) [inline pages/pipelines/templates.md:21]: I don't think the "also" fits well in the numbered steps.

```suggestion
1. Select _Edit_ if you want to change the step configuration for the template.
```

**Review by mbelton-buildkite — APPROVED** [review]
A couple of final tweaks to consider, but looks good to me 😊

**Comment by mhaylock** [comment]
Thanks @mbelton-buildkite.

Regards these two items I mentioned as missing in the description, there will be a seperate PR to update the GraphQL docs, and another to update the REST API docs:

> Missing:
>
> Updated GraphQL docs (should have this ready tomorrow)
> An example in the REST API docs (not sure if this is needed?)

**Pull request merged**
